pkg/controller/service/service_controller.go

```diff
@@ -582,6 +591,38 @@ func nodeSlicesEqualForLB(x, y []*v1.Node) bool {
 	return nodeNames(x).Equal(nodeNames(x))
 }
 
+// getNodeTaint Get node based on Taint
+func getNodeTaint() corelisters.NodeConditionPredicate {
```

Comment:
We might want to change naming for the fun we return, e.g go from `corelisters.NodeConditionPredicate` to something like `corelisters.TaintPredicate`

Reply:
I agree to rename `corelisters.NodeConditionPredicate` but we can using a generic name as :  `corelisters.NodePredicate` and refacto to avoid duplication of code in `node_expansion.go`

Reply:
sure